### PR TITLE
integrated feedback & tests for views/services

### PR DESCRIPTION
**Pull request by abhn**
- <strike>todo - testing is becoming a bottleneck.</strike>
- <strike>Need to separate model fetching login into services.</strike>

### CODE REVIEW

File: app/services.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+def add_task(current_user_id, task):
+    """get user_id and task and add that task for that user. Assume authentication"""
+
+    from .models import Tasks
+
+    task = Tasks(mentee_id=current_user_id, task=task)
+
+    db_init.session.add(task)
+    db_init.session.commit()
```
Comment (darth-dodo) [inline app/services.py]: what happens in case of transactional error?

File: app/services.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+
+def add_mentor(mentor_name, mentor_email, current_user_id):
+    from .models import Mentors, Mentees
+
+    # something's fishy, return
+    if not mentor_name or not mentor_email:
+        return
+
+    mentee_present = Mentees.query.filter_by(id=current_user_id).first()
```
Comment (darth-dodo) [inline app/services.py]: assuming `id is the pk, shouldn't this be get or 404

Comment (abhn) [reply app/services.py]: That case is also handled in the views (and apis) `if add_mentor(...): do x`

Raising 404 will send a 404 to the user which makes up for inconsistent user experience and leak info. This way, I'm able to keep a generic interface for failures `{'success': False}, 500`. What would you suggest?

**Comment by darth-dodo** [comment]
Raise custom business exceptions using decorators

**Pull request merged**